### PR TITLE
Add ability to audit/update azure-cli setup.py in `azdev verify version` command

### LINKED ISSUE #31: `azdev verify version` should populate the azure-cli setup.py dependencies and/or verify them
Two modes:
1) scrape the versions from the modules into azure-cli's setup.py file: this is to minimize the manual burden of copying versions directly and will ensure that all modules in the CLI make it into the setup.py file.
2) verify the versions of each module match those in setup.py. This would be used by the release pipeline as a final verification before continuing.

### PR DESCRIPTION
**Pull request by tjprescott**
Closes #31.

New usage:
- `azdev verify version --update` - will verify local versions against PyPI AND update setup.py with the current versions. This would be run on the release branch to bump necessary module versions and ensure setup.py for azure-cli reflects those versions.

- `azdev verify version <MOD>` - will verify local versions against PyPI but do nothing with azure-cli setup.py file.

- `azdev verify version` - will verify local versions against PyPI and audit local versions against the azure-cli setup.py file. This would be run as a precheck in the publish phase.

### CODE REVIEW

File: azdev/operations/pypi.py [hunk 1]
```diff
@@ -261,36 +267,109 @@ def verify_versions(modules=None):
...
+    # only audit or update setup.py when all modules are being considered
+    # otherwise, edge cases could arise
+    if no_update is None:
+        return results
+
+    # retrieve current versions from azure-cli's setup.py file
+    azure_cli_path = get_path_table(include_only='azure-cli')['core']['azure-cli']
+    azure_cli_setup_path = find_files(azure_cli_path, SETUP_PY_NAME)[0]
+    with open(azure_cli_setup_path, 'r') as f:
+        setup_py_version_regex = re.compile(r"'([^'=]*)==([\d.]*)'")
```
Comment (marstr) [inline azdev/operations/pypi.py]: In general, regex compilation is relatively expensive. You may be able to get some perf improvements by moving this into __init__ or something else that will only execute once at global scope.

Comment (marstr) [inline azdev/operations/pypi.py]: Looks like this pattern only supports python strings that are defined with single-ticks at the moment. It'd be pretty easy to updated it to support single or double ticks.

Comment (marstr) [inline azdev/operations/pypi.py]: Another small thing with regex, I always prefer named groups. IMHO, they're more readable, and more resilient to changes to the pattern itself later.

File: azdev/help.py [hunk 2]
```diff
@@ -81,7 +81,16 @@
...
+    short-summary: Verify package versions against those hosted on PyPI and/or in the azure-cli setup.py file.
+    long-summary: >
+        This is used to ensure the correct module versions are bumped prior to release and that all
+        module versions are present in the azure-cli setup.py with their versions pinned.
+    examples:
+        - name: Verify all versions and update azure-cli's setup.py with each module's verison.
+          text: azdev verify version
+
+        - name: Verify all versions and audit them against azure-cli's setup.py ONLY.
+          text: azdev verify version --no-update
```
Comment (marstr) [inline azdev/help.py]: Nit, I'd still prefer that the default action was to just audit here, and that providing a flag that said "overwrite" or something similar caused changes to disk. Makes more sense to me when the command is "verify". That said, I with Git and the CI scenarios that we have in mind, it's no big deal.

Comment (tjprescott) [reply azdev/help.py]: I figured to save you some typing, but I agree with you in principle. I'll make the change :)

File: azdev/operations/pypi.py (hunk 1 above)
Comment (tjprescott) [reply azdev/operations/pypi.py]: Well, this will only get executed once anyways, but only if you run this command.

Comment (tjprescott) [reply azdev/operations/pypi.py]: Good point. I did it based on the current format, but I can make it more robust.

**Pull request merged**